File: utilslib/src/main/java/com/ashlikun/utils/ActivityLifecycleCallbacksAdapter.java
package com.ashlikun.utils;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

/**
 * 作者　　: 李坤
 * 创建时间: 2019/4/18　14:38
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：对{@link Application.ActivityLifecycleCallbacks} 的简化
 */
public abstract class ActivityLifecycleCallbacksAdapter implements Application.ActivityLifecycleCallbacks {
    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(Activity activity) {

    }

    @Override
    public void onActivityResumed(Activity activity) {

    }

    @Override
    public void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(Activity activity) {

    }
}

File: utilslib/src/main/java/com/ashlikun/utils/ActivityLifecycleCallbacksAdapter.kt
package com.ashlikun.utils

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.15 15:53
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：对[Application.ActivityLifecycleCallbacks] 的简化
 */

abstract class ActivityLifecycleCallbacksAdapter : ActivityLifecycleCallbacks {
    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle) {}
    override fun onActivityStarted(activity: Activity) {}
    override fun onActivityResumed(activity: Activity) {}
    override fun onActivityPaused(activity: Activity) {}
    override fun onActivityStopped(activity: Activity) {}
    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
    override fun onActivityDestroyed(activity: Activity) {}
}